### ISSUE TITLE
Add auto reloading by database file updating

### ISSUE THREAD

**Issue by marataziat** [sqlitebrowser/sqlitebrowser#1767]
Can you make automatical "refresh data in the selected table" like when I write tail -f it will show me all updates on file live. Can you add this to sqlitebrowser?

**Comment by chrisjlocke** [comment]
You can just press <kbd>F5</kbd>.  Regularly refreshing the database would 'refresh' the grid as well - so if you were scrolling through the grid, you'd be back at the beginning.  Also, it would have to be a user-configured time period - do you want 5 minutes, 1 minute, 1 second?  
DB4S wouldn't be able to know if records had changed or been updated.  It might be able to look at the database time stamp or table record count number, but for large tables, that can be a costly exercise.  Looking at the database time stamp would tell DB4S the database has been updated, but not which table.
Which operating system are you using?  There are 'macro recorders' so for example, a program can grab the focus of DB4S then 'press' the <kbd>F5</kbd> key.

**Comment by karim** [comment]
What about auto reloading the active table when the opened database file is modified?

DB4S can set a file watcher, using [QFileSystemWatcher](https://doc.qt.io/qt-5/qfilesystemwatcher.html), to monitor the database file when it is opened. If the file is modified then DB4S can assume that the database file was modified by other process.

Next, DB4S run [`PRAGMA data_version`](https://sqlite.org/pragma.html#pragma_data_version) to be sure that there were committed changes made by other connection/process, then it reloads the current active table.

This will prevent the constant polling to check if there were any changes to the database, and allow using DB4S as a *live viewer* (on-top/active on screen) while the database is being updated in the background by other program/script.

A toggle also can be used to enable/disable reloading data when they are changed by other process will be useful.

**Comment by chrisjlocke** [comment]
> What about auto reloading the active table when the opened database file is modified?

But you don't know what table was modified.  I have a database that holds activity logs, so is constantly being refreshed.  I may want to monitor the sausage roll table, which is only updated infrequently (shame, as I like sausage rolls...)  DB4S will constantly refresh the grid unnecessarily.
Pressing F5 may not also work, as you have to scroll the grid to the right record - again, depending on the sort applied.  Auto updating a SQL tab might be better approach - you can set a query to be specific, 'no rows found' - AUTO REFRESH - 'pop!' now it exists....

**Comment by karim** [comment]
The use case that I had in mind is.. you are already modifying the database file using the command line/script/other tool, while opening the file to view it using a GUI app (DB4S) to check for changes, so you are likely selecting the table being modified as the active tab, and seeing the changes live.

>  I have a database that holds activity logs, so is constantly being refreshed. I may want to monitor the sausage roll table, which is only updated infrequently (shame, as I like sausage rolls...) DB4S will constantly refresh the grid unnecessarily.

This is another use case that is also common. At the moment, there is no way of knowing if a specific table is modified. Maybe we can set/unset the tables to be refreshed using the project files?

**Comment by chrisjlocke** [comment]
I think that's over-engineering it and adding extra coding complexity.... but just my thought.  Maybe its easy to implement...

**Comment by mgrojo** [comment]
This feature makes sense and it would be useful, but it's true that presents some challenges. I'll add a reference of a previous issue where this was discussed:  #1327.

**Comment by tigran123** [comment]
Yes, it would make perfect sense as it would be very useful. I was looking for this feature as switch the focus back to and forth between SQLiteStudio and other applications and pressing `F5` is not very convenient, even though I set 'focus follows mouse' on my development workstation.

**Comment by chrisjlocke** [comment]
> I was looking for this feature

For what .. just viewing a set of records?

**Comment by tigran123** [comment]
> For what .. just viewing a set of records?

Yes, the application I am developing (a ChatGPT-like system, but much more flexible and also handling LaTeX properly and consistently) is modifying several tables and I want to see these modifications in real-time. Btw, it modifies two tables, so the ability to view records in more than one table simultaneously would also be nice.

**Comment by tigran123** [comment]
Oh dear, I think I am commenting in the wrong repository -- I am talking about SQLiteStudio and this github repository is about a different prograrm, called sqlitebrowser! Maybe I should try this one :)

**Comment by GribanovIvan** [comment]
> > I was looking for this feature
> 
> For what .. just viewing a set of records?

As a software engineer student, I could also say that that feature is not useless. I'm using your software to learn about databases, doing random stuff and wait immediately see if it exploded or not. I know that not new software may be hard to modify but what if you just add inotify to it and when it triggered do refresh as if corresponding button was pressed? Of course, not by default.

**Comment by mgrojo** [comment]
This is implemented on the development branch.

Automatically refresh read-only databases on modified database file

This adds a new setting in "Preferences > Database" (defaulting to false)
to watch the database file when opened in read-only mode and refresh the 
browser on updated file.

If someone want to test it and report any problem, use the next nightly or continuous build.
